larray/core/array.py

```diff
@@ -1821,12 +1826,17 @@ def _translate_axis_key_chunk(self, axis_key, bool_passthrough=True):
 
         # labels but known axis
         if isinstance(axis_key, LGroup) and axis_key.axis is not None:
-            real_axis = self.axes[axis_key.axis]
             try:
-                axis_pos_key = real_axis.index(axis_key, bool_passthrough)
+                real_axis = self.axes[axis_key.axis]
+                try:
+                    axis_pos_key = real_axis.index(axis_key, bool_passthrough)
+                except KeyError:
+                    raise ValueError("%r is not a valid label for any axis" % axis_key)
+                return real_axis.i[axis_pos_key]
             except KeyError:
-                raise ValueError("%r is not a valid label for any axis" % axis_key)
-            return real_axis.i[axis_pos_key]
+                # axis associated with axis_key may not belong to self.
+                # In that case, we need to search for a compatible axis as below
+                pass
```

Comment:
I think this case only works in your test by accident. If I understand correctly, in that case, the axis_key will be left as-is with an axis which does not correspond to any axis from self.axes. But the code from axis.index (used below) has a test which should make this fail:
```python
            if isinstance(key.axis, basestring) and key.axis != self.name:
                raise KeyError(key)
```
I think you need to replace the "pass" here with something like axis_key = axis_key.to_label()

I think the reason it seems to work is the "potential_tick = _to_tick(key)" code block in Axis.index which actually gives the same string in this case. You probably want to add a test with a list of ticks for which _to_tick would not return the same string than the actual tick.